[NPU][Wan2.2-S2V] Inference fails: aclnnIndex does not support DT_COMPLEX64 in WanS2VRotaryPosEmbed

### Your current environment

### Environment

| Component | Version |
|-----------|---------|
| Hardware | 2× Ascend 910B3 (64 GB HBM each) |
| Docker image | `m.daocloud.io/quay.io/ascend/vllm-omni:v0.23.0rc1` |
| vLLM-Omni | `0.23.0rc2.dev147+g724f5d132` (commit `724f5d132deb55dfd73591c83494eca15c95d179`) |
| vLLM | `0.23.0` |
| vLLM-Ascend | `0.19.1rc2.dev536+g4885d8e7b` |
| PyTorch | `2.10.0+cpu` |
| torch_npu | `2.10.0` |
| cache-dit | `1.3.0` |
| Python | 3.12 |
| Model | `Wan-AI/Wan2.2-S2V-14B` (local path: `/data/Wan-AI/Wan2.2-S2V-14B`) |

### Your code version

<details>
<summary>The commit id or version of vllm</summary>

```text

```
</details>
<details>
<summary>The commit id or version of vllm-omni</summary>

```text

```
</details>


### 🐛 Describe the bug

### Summary

Wan2.2-S2V (`Wan22S2VPipeline`) can **start successfully** on Ascend NPU, but **inference always fails** on the first denoising step with `aclnnIndex` error: `DT_COMPLEX64` is not supported. The failure occurs in `WanS2VRotaryPosEmbed.forward()` when indexing a `torch.complex64` frequency tensor on NPU.

This blocks all S2V generation on Ascend, including online serving via `POST /v1/videos/sync`.


### Steps to Reproduce

1. Start the S2V server on Ascend NPU with TP=2:

```bash
VLLM_USE_MODELSCOPE=True \
VLLM_WORKER_MULTIPROC_METHOD=spawn \
vllm serve Wan-AI/Wan2.2-S2V-14B --omni \
  --model-class-name WanS2VPipeline \
  --tensor-parallel-size 2 \
  --flow-shift 3.0 \
  --vae-use-slicing --vae-use-tiling \
  --cache-backend cache_dit \
  --port 8200


Expected Behavior
Video generation completes and returns an MP4 file.

Actual Behavior
Server starts fine; API accepts the request.
Inference fails at step 0/40 during RoPE precomputation.
HTTP 500 Internal Server Error is returned.
Both TP ranks (Device 0 and Device 1) fail with the same error.

Error Log
RuntimeError: index_high_dims_op_api:../third_party/op-plugin/op_plugin/ops/opapi/IndexKernelNpuOpApi.cpp:40
NPU function error: call aclnnIndex failed, error code is 161002
[ERROR] ERR00100 PTA call acl api failed.
AclNN_Parameter_Error(EZ1001): Tensor self not implemented for DT_COMPLEX64,
should be in dtype support list [DT_FLOAT,DT_FLOAT16,DT_INT64,DT_INT32,DT_BOOL,
DT_INT8,DT_UINT8,DT_BFLOAT16,DT_DOUBLE,DT_INT16,].

Stack trace (abbreviated):
pipeline_wan2_2_s2v.py:1271  forward → diffuse()
pipeline_wan2_2_s2v.py:1029  diffuse → predict_noise_maybe_with_cfg()
pipeline_wan2_2_s2v.py:944   predict_noise → current_model(**kwargs)
wan2_2_s2v_transformer.py:1614  forward → prepare_rope()
wan2_2_s2v_transformer.py:1329  prepare_rope → self.rope(x.detach(), grid_sizes)
rope.py:400  WanS2VRotaryPosEmbed.forward → freqs_split[0][f_sam]

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

--- a/vllm_omni/diffusion/layers/rope.py
+++ b/vllm_omni/diffusion/layers/rope.py
@@ -423,16 +423,16 @@ class WanS2VRotaryPosEmbed(torch.nn.Module):
                         h_sam = torch.linspace(int(h_o), int(t_h + h_o) - 1, seq_h_int, device=device).long()
                         w_sam = torch.linspace(int(w_o), int(t_w + w_o) - 1, seq_w_int, device=device).long()
 
-                        freqs_0 = freqs_split[0][f_sam] if f_o >= 0 else freqs_split[0][f_sam].conj()
+                        freqs_0 = torch.index_select(freqs_split[0] if f_o >= 0 else freqs_split[0].conj(), 0, f_sam)
                         freqs_0 = freqs_0.view(seq_f_int, 1, 1, -1)
 
                         freqs_i = torch.cat(
                             [
                                 freqs_0.expand(seq_f_int, seq_h_int, seq_w_int, -1),
-                                freqs_split[1][h_sam]
+                                torch.index_select(freqs_split[1], 0, h_sam)
                                 .view(1, seq_h_int, 1, -1)
                                 .expand(seq_f_int, seq_h_int, seq_w_int, -1),
-                                freqs_split[2][w_sam]
+                                torch.index_select(freqs_split[2], 0, w_sam)
                                 .view(1, 1, seq_w_int, -1)
                                 .expand(seq_f_int, seq_h_int, seq_w_int, -1),
                             ],
@@ -523,14 +523,18 @@ class RotaryEmbeddingS2VGrid(torch.nn.Module):
                         h_sam = torch.linspace(int(h_o), int(t_h + h_o) - 1, seq_h_int, device=device).long()
                         w_sam = torch.linspace(int(w_o), int(t_w + w_o) - 1, seq_w_int, device=device).long()
 
-                        freqs_0 = freqs[0][f_sam] if f_o >= 0 else freqs[0][f_sam].conj()
+                        freqs_0 = torch.index_select(freqs[0] if f_o >= 0 else freqs[0].conj(), 0, f_sam)
                         freqs_0 = freqs_0.view(seq_f_int, 1, 1, -1)
 
                         freqs_i = torch.cat(
                             [
                                 freqs_0.expand(seq_f_int, seq_h_int, seq_w_int, -1),
-                                freqs[1][h_sam].view(1, seq_h_int, 1, -1).expand(seq_f_int, seq_h_int, seq_w_int, -1),
-                                freqs[2][w_sam].view(1, 1, seq_w_int, -1).expand(seq_f_int, seq_h_int, seq_w_int, -1),
+                                torch.index_select(freqs[1], 0, h_sam)
+                                .view(1, seq_h_int, 1, -1)
+                                .expand(seq_f_int, seq_h_int, seq_w_int, -1),
+                                torch.index_select(freqs[2], 0, w_sam)
+                                .view(1, 1, seq_w_int, -1)
+                                .expand(seq_f_int, seq_h_int, seq_w_int, -1),
                             ],
                             dim=-1,
                         ).reshape(seg_len, 1, -1)

